pkg/network/ovn_kubernetes.go

```diff
@@ -861,14 +870,66 @@ func validateOVNKubernetes(conf *operv1.NetworkSpec) []error {
 				if oc.V6InternalSubnet != "" {
 					_, v6ServiceNet, _ := net.ParseCIDR(sn)
 					if iputil.NetsOverlap(*v6Net, *v6ServiceNet) {
-						out = append(out, errors.Errorf("v6InternalSubnet overlaps with ServiceNetwork %s", sn))
+						out = append(out, errors.Errorf("v6InternalSubnet %s overlaps with ServiceNetwork %s", oc.V6InternalSubnet, sn))
 					}
 				}
 			} else {
 				if oc.V4InternalSubnet != "" {
 					_, v4ServiceNet, _ := net.ParseCIDR(sn)
 					if iputil.NetsOverlap(*v4Net, *v4ServiceNet) {
-						out = append(out, errors.Errorf("v4InternalSubnet overlaps with ServiceNetwork %s", sn))
+						out = append(out, errors.Errorf("v4InternalSubnet %s overlaps with ServiceNetwork %s", oc.V4InternalSubnet, sn))
+					}
+				}
+			}
+		}
+		if oc.GatewayConfig != nil {
+			if oc.GatewayConfig.IPv4.InternalMasqueradeSubnet != "" {
+				if !cnHasIPv4 {
+					out = append(out, errors.Errorf("v4InternalMasqueradeSubnet %s and ClusterNetwork must have matching IP families", oc.GatewayConfig.IPv4.InternalMasqueradeSubnet))
+				}
+				_, v4Net, err = net.ParseCIDR(oc.GatewayConfig.IPv4.InternalMasqueradeSubnet)
+				if err != nil {
+					out = append(out, errors.Errorf("v4InternalMasqueradeSubnet is invalid: %s", err))
+				}
```

Comment:
I see in other places that the error returned from `net.ParseCIDR` is ignored, I suspect because it is assumed to be properly validated by the API server. But here you don't. If there is an error here, are you sure you can derenference safely `*v4Net` later on?

Reply:
correct the api would catch any invalid subnet configuration (or at least it should), just being cautious here to catch anything that i missed with my api validations. I am unsure if a value that could cause a panic would occur here because we already check for nil

Reply:
Where do you check for nil?

Reply:
we check that the gatewayConfig is not nil, so if no value is passed into the internalMasqueradeSubnet it should just be an empty string im pretty sure

Reply:
So you have
```
_, v4MasqNet, err = net.ParseCIDR(oc.GatewayConfig.IPv4.InternalMasqueradeSubnet)
```
if that returns non-nil `err` and potentially nil `v4MasqNet` then you just append the error but continue on
```
out = append(out, errors.Errorf("v4InternalMasqueradeSubnet is invalid: %s", err)
```
and then a few lines below you dereference v4MasqNet which will panic if it is nil
```
if iputil.NetsOverlap(*v4MasqNet, *v4ClusterNet)
```

Reply:
@jcaamano added a check to make sure that the v4/v6 masqNet return isnt nil